### PR TITLE
[Modules] Updated Compute/AvailabilitySets to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Compute/AvailabilitySets to new dependency approach

| Pipeline |
| - |
| [Compute: AvailabilitySets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.availabilitysets.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Compute/availabilitySets/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,56 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Compute/availabilitySets/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Compute/availabilitySets/.test/min/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,37 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Compute/availabilitySets/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Compute/availabilitySets/.test/common/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+resource proximityPlacementGroup 'Microsoft.Compute/proximityPlacementGroups@2022-03-01' = {
+    name: proximityPlacementGroupName
+    location: location
+}
+
+@description('The principal ID of the created Managed Identity.')
+output managedIdentityPrincipalId string = managedIdentity.properties.principalId
+
+@description('The resource ID of the created Managed Identity.')
```
Comment (eriqua) [inline modules/Microsoft.Compute/availabilitySets/.test/common/dependencies.bicep:23]: Please remove output if not used

Comment (MrMCake) [reply modules/Microsoft.Compute/availabilitySets/.test/common/dependencies.bicep]: done

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**